pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go

```diff
@@ -423,17 +451,127 @@ func TestWithBinding(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			state := framework.NewCycleState()
 			node := &framework.NodeInfo{}
 			node.SetNode(test.Node)
 			p := &VolumeZone{
 				pvLister,
 				pvcLister,
 				scLister,
 			}
-			gotStatus := p.Filter(context.Background(), nil, test.Pod, node)
-			if !reflect.DeepEqual(gotStatus, test.wantStatus) {
-				t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+			_, gotPreFilterStatus := p.PreFilter(ctx, state, test.Pod)
+			if !gotPreFilterStatus.IsSuccess() {
+				if !reflect.DeepEqual(gotPreFilterStatus, test.wantStatus) {
+					t.Errorf("status does not match: %v, want: %v", gotPreFilterStatus, test.wantStatus)
+				}
+			} else {
+				gotStatus := p.Filter(ctx, state, test.Pod, node)
+				if !reflect.DeepEqual(gotStatus, test.wantStatus) {
+					t.Errorf("status does not match: %v, want: %v", gotStatus, test.wantStatus)
+				}
 			}
 		})
 	}
 }
+
+func BenchmarkVolumeZone(b *testing.B) {
+	tests := []struct {
+		Name       string
+		Pod        *v1.Pod
+		NumPV      int
+		NumPVC     int
+		NumNodes   int
+		wantStatus *framework.Status
+	}{
+		{
+			Name:     "zone label matched",
```

Comment:
Do you plan to add more cases? If not, do not create a test table.

Reply:
> Do you plan to add more cases? If not, do not create a test table.

After this update, I've created two cases for Benchmark, one with prefilter called, one without. So in first case, we calculate pv only once, in second case, we calculate once for each node.